Dockerfile.prod

```diff
@@ -0,0 +1,97 @@
+# MIT License
+#
+# (C) Copyright 2022-2023 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+# STAGE 1
+FROM       artifactory.algol60.net/docker.io/library/alpine:3.17 as builder
+RUN        apk add --update --no-cache \
+             py3-pip \
+             make \
+             cmake \
+             gcc \
+             g++ \
+             openssl \
+             jq \
+             vim \
+             libffi-dev \
+             musl-dev \
+             python3 \
+             python3-dev \
+             git
+# make canu group, create (system) canu user, add to canu group, make home folder, and set login shell, no password
+RUN        addgroup -S canu 
+RUN        adduser \
+             -S canu \
+             -G canu \
+             -h /home/canu \
+             -s /bin/bash \
+             -D
+USER       canu
+RUN        mkdir -p /home/canu/app/wheels
+WORKDIR    /home/canu/app
+# copy the source code into the container
+COPY       --chown=canu:canu . .
+# Install the build dependencies as wheels
+ENV        PYTHONDONTWRITEBYTECODE 1
+ENV        PYTHONUNBUFFERED 1
+RUN        python -m pip install --user -U pyinstaller wheel setuptools
```

Comment:
Instead of installing these by hand, use the build module to read what's defined in `pyproject.toml`.

```bash
python -m pip install -U build
python -m build -q
```

Then either install `pyinstaller` by hand, or add it to the build-system requirements.

Reply:
These steps are in stage1 of the build, and never make it into the final image.  This reduces the image size and removes cruft from the final image by only installing the wheels from stage 1: https://github.com/Cray-HPE/canu/pull/255/files#diff-a811e274b83c4976b7b48589ed3273262e4adb67534b49208d484fe7f0bc06a0R84

I was working off what was in the spec file originally, but I see that has also changed.  Despite that, this works well as-is.  Is it blocking for you?

Reply:
I tried this in any case to see.  It works fine in stage1, but this adds additional dependencies into the final image, `build` and `wheel` and there are still errors getting it installed.

I am far from a python packaging expert, but the approach I have in here now only requires a `pip install` pointed at the copied over wheels from the first stage.

Reply:
Here is what we came to:

Specific to canu-in-a-container, `wheel` will be used to reduce the final image size by building the dependencies in stage 1, saving them as `.whl`s and copying them into stage2.  This differs slightly from the rpm install, but greatly improves the image size and still gets the same end result.